src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointIndexer.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.mapper.xypoint;
+
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYPoint;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import lombok.AllArgsConstructor;
+
+import org.apache.lucene.document.XYPointField;
+import org.apache.lucene.geo.XYPoint;
+import org.apache.lucene.index.IndexableField;
+import org.opensearch.geometry.Point;
+import org.opensearch.index.mapper.AbstractGeometryFieldMapper;
+import org.opensearch.index.mapper.ParseContext;
+
+/**
+ * Converts points into Lucene-compatible form for indexing in a xy_point field.
+ */
+@AllArgsConstructor
+public class XYPointIndexer implements AbstractGeometryFieldMapper.Indexer<List<ParsedXYPoint>, List<? extends XYPoint>> {
+    private final String fieldName;
+
+    @Override
+    public List<? extends XYPoint> prepareForIndexing(List<ParsedXYPoint> points) {
+        Objects.requireNonNull(points, "XYPoint cannot be null");
```

Comment:
why not use @NonNull in the argument parameter

Reply:
ack